Sources/FoundationEssentials/FileManager/FileOperations.swift

```diff
@@ -817,7 +817,16 @@ enum _FileOperations {
             guard delegate.shouldPerformOnItemAtPath(src, to: dst) else { return }
 
             try dst.withNTPathRepresentation { pwszDestination in
-                if faAttributes.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY {
+                // Check for reparse points first because symlinks to directories are reported as both reparse points and directories, and we should copy the symlink not the contents of the linked directory
+                if faAttributes.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT == FILE_ATTRIBUTE_REPARSE_POINT {
```

Comment:
Would a non-symlinked directory happen to satisfy this constraint as well, and is that the desired behavior? By reading at this code, I'm not sure if `fileManager.createSymbolicLink` would succeed since it's not a symlink. And if it fails, we'd `throw` instead of proceeding to copy the content of the directory. Is that ok?

Reply:
Yeah plain directories won't have this flag set, but other types of reparse points that aren't supported by `destinationOfSymbolicLink` would likely fail here - I think we can address that later since most of the `FileManager` code in general likely doesn't do well with other types of reparse points